pwnlib/term/key.py

```diff
@@ -18,7 +18,7 @@
 FLAG_CONVERTKP = True
 
 try:    _fd = sys.stdin.fileno()
-except Exception: _fd = file('/dev/null', 'r').fileno()
+except Exception: _fd = os.open(os.devnull, os.O_RDONLY).fileno()
```

Comment:
`os.open` already returns a file number (your current code results with `AttributeError` on an `int` having no `fileno`).

Just to mention, the previous `file(...).fileno()` had no sense whatsoever, since the object that held the file number open kept being destroyed (and the `__del__` finalizer was calling `close` by the time anything interesting happened), so it ended up using an invalid file descriptor (this path was surely not covered by tests BTW).

```suggestion
except Exception: _fd = os.open(os.devnull, os.O_RDONLY)
```